deps/wazuh_testing/wazuh_testing/dashboard_saturation_tests/dashboard_saturation_tests.py

```diff
@@ -0,0 +1,217 @@
+import argparse
+import json
+import pandas as pd
+
+from datetime import datetime
+from os import makedirs
+from os.path import exists
+from subprocess import run
+
+# Global Configuration Variables
+csv_types = ['aggregate', 'intermediate']
+url_format = 'https://'
+
+def format_directory(directory):
+    """Check the paths of the directories and format them if necessary.
+    Args:
+        args (string): Path of a directory.
+    Returns:
+        string: Formatted path.
+    """
+    return directory if directory.endswith("/") else f'{directory}/'
+
+def check_csv_types(types):
+    """Check that the CSV type is valid.
+    Args:
+        types (list): List of types.
+    """
+    for value in types:
+        if (value not in csv_types):
+            raise Exception("CSV type is not valid") 
+        
+def create_directories(directory):
+    """Create directory that does not exist.
+    Args:
+        directory (str): Path of directory.
+    """
+    if not exists(directory):
+        makedirs(directory)
+
+def create_session_file(path, user):
+    """Create file to save the browser session.
+    Args:
+        path (str): Path of directory.
+        user (str): Username.
+    """
+    file_path = f"{path}user-{user}.json"
+
+    with open(file_path, 'w') as file:
+        file.write('{}')
+
+def process_script_arguments(args):
+    """Process script arguments, create folders and generate necessary files.
+    Args:
+        args (ArgumentParser): Script parameters.
+    """
+    # Format Directories
+    args.logs = format_directory(args.logs)
+    args.screenshots = format_directory(args.screenshots)
+    args.csv = format_directory(args.csv)
+    args.session = format_directory(args.session)
+
+    # Check CSV Types
+    check_csv_types(args.type)
+
+    # Create Directories
+    create_directories(args.logs)
+    create_directories(args.screenshots)
+    create_directories(args.csv)
+    create_directories(args.session)
+
+    # Create Session File
+    create_session_file(args.session, args.user)
+
+def format_artillery_params(key, value):
+    """Format options for Artillery.
+    Args:
+        key (str): Name of the Artillery option.
+        value (str): Value of the Artillery option.
+    Returns:
+        str: Formatted Artillery option.
+    """
+    return f"\"{key}\": \"{value}\""
+
+def gen_artillery_params(args):
+    """Format all parameters for Artillery.
+    Args:
+        args (ArgumentParser): Script parameters.
+    Returns:
+        str: Formatted Artillery parameters.
+    """
+    user = format_artillery_params('username', args.user)
+    password = format_artillery_params('password', args.password)
+    screenshots = format_artillery_params('screenshots', args.screenshots)
+    session = format_artillery_params('session', args.session)
+
+    params = f"\'{{{user}, {password}, {screenshots}, {session}}}\'"
+
+    return params
```

Comment:
Consider launching artillery via a configuration file instead of passing parameters directly. This approach will help avoid error-prone logic